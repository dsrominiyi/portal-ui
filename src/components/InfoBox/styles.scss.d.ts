export const infoBox: string;
export const topContainer: string;
export const headerContainer: string;
export const header: string;
export const moreAction: string;
export const subHeader: string;
export const headerDivide: string;
export const useDivider: string;
export const itemList: string;
export const item: string;
