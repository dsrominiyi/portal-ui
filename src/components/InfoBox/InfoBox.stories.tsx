import React, { ReactNode } from 'react';
import { storiesOf } from '@storybook/react';
import MoreActionsLink from '../IconLink/MoreActionsLink/index';
import { PolicyType } from '../../helpers/policyTypeInfoMap';

import PolicyWidget from '../PolicyWidget';
import InfoBox from '.';

const stories = storiesOf('InfoBox', module);

const defaultItems: ReactNode[] = [];

for (let i = 0; i < 10; i += 1) {
  const policyWidgetProps = {
    id: `PN000${i}`,
    type: PolicyType.PrivateCar
  };
  defaultItems.push(<PolicyWidget {...policyWidgetProps} />);
}

const render = (
  items?: ReactNode[],
  header?: string,
  useDivider?: boolean,
  subHeader?: string,
  moreActions?: ReactNode[],
  hideHeaderDivider?: boolean
) => () => (
  <InfoBox
    items={items}
    header={header}
    useDivider={useDivider}
    subHeader={subHeader}
    moreActions={moreActions}
    hideHeaderDivider={hideHeaderDivider}
  />
);

export const storyRenders = {
  titleAndDivider: render(defaultItems, 'Policies', true),
  noDivider: render(defaultItems, 'Policies'),
  noTitle: render(defaultItems, undefined),
  noItems: render(undefined, 'Policies'),
  subHeaderAndMoreActions: render(defaultItems, 'Policies', true, 'Private Car', [
    <MoreActionsLink key="moreActionsLink" onClick={() => alert('You clicked!')} />
  ]),
  noHeaderDivider: render(defaultItems, 'Policies', true, undefined, undefined, true)
};

stories.add('Title and divider', storyRenders.titleAndDivider);
stories.add('Without divider', storyRenders.noDivider);
stories.add('Without title', storyRenders.noTitle);
stories.add('Without items', storyRenders.noItems);
stories.add('Sub Header and MoreActions', storyRenders.subHeaderAndMoreActions);
stories.add('No Header Divider', storyRenders.noHeaderDivider);
