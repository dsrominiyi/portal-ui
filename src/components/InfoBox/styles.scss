@import '../../style/vars';

$divider-colour: transparentize($brand-black, 0.9);

.info-box {
  display: inline-flex;
  flex-direction: column;
  overflow: visible;
  height: auto;
  background-color: $brand-white;
  font-size: 12px;
  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.08);
  padding: 0 1.25rem;
}

.top-container {
  padding: 1.25rem 0;
}

.header-container {
  display: flex;

  > * {
    flex: 1;
  }
}

.header {
  display: flex;
  justify-content: flex-start;
  color: $brand-charcoal;
  font-size: 1.5rem;
  font-weight: 600;
  line-height: 1.2;
}

.more-action {
  display: flex;
  justify-content: flex-end;
}

.sub-header {
  margin-top: 0.5rem;
  font-size: 1.2rem;
  color: $brand-charcoal;
}

.header-divide {
  border-bottom: 1px $divider-colour solid;
}

.use-divider {
  border-bottom: 1px $divider-colour solid;

  &:last-child {
    border-bottom: 0;
  }
}

.item-list {
  display: flex;
  flex-direction: column;
  width: 100%;

  .item {
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 1.25rem 0;

    > * {
      flex: 1;
    }
  }
}
