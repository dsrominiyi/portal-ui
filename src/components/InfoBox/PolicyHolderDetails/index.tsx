import React, { FunctionComponent, MouseEvent } from 'react';
import { utc as moment } from 'moment';

import MoreActionsLink from '../../IconLink/MoreActionsLink';
import InfoBox from '..';

import styles from './styles.scss';

export interface CustomerDetails {
  name: string;
  dateOfBirth?: string;
  address: Address;
  contactDetails: Contact[];
}

export interface Props {
  id: string;
  isQuote: boolean;
  policyType: string;
  customerDetails: CustomerDetails;
  onClickAction: (event: MouseEvent) => void;
}

const PolicyHolderDetails: FunctionComponent<Props> = ({
  id,
  isQuote,
  policyType,
  customerDetails,
  onClickAction
}: Props) => {
  const address = Object.values(customerDetails.address)
    .filter(val => val !== undefined && val !== '')
    .join(', ');

  const contacts = customerDetails.contactDetails.map((contact, index) => {
    const newLine = index !== customerDetails.contactDetails.length - 1 ? '\n' : '';
    return `${contact.type[0].toUpperCase() + contact.type.slice(1)}: ${contact.detail}${newLine}`;
  });

  const body = (
    <div className={styles.policyHolderDetails}>
      <p className={styles.header}>Policyholder</p>
      <p className={styles.subHeader}>{customerDetails.name}</p>
      {customerDetails.dateOfBirth && (
        <p className={styles.textSection}>
          {moment(customerDetails.dateOfBirth).format('DD/MM/YYYY')}
        </p>
      )}
      <p className={styles.subHeader}>Address:</p>
      <p className={styles.textSection}>{address}</p>
      <p className={styles.subHeader}>Contact:</p>
      <p className={styles.textSection}>{contacts}</p>
    </div>
  );

  return (
    <InfoBox
      header={`${isQuote ? 'Quote' : 'Policy'}: ${id}`}
      subHeader={policyType}
      moreActions={[<MoreActionsLink key="moreActionsLink" onClick={onClickAction} />]}
      items={[body]}
      additionalHeaderClasses={[styles.infoBoxHeader]}
    />
  );
};

PolicyHolderDetails.displayName = 'PolicyHolderDetails';
export default PolicyHolderDetails;
