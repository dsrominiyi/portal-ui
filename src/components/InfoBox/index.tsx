import React, { FunctionComponent, ReactNode } from 'react';
import classNames from 'classnames/bind';

import styles from './styles.scss';

interface Props {
  items?: ReactNode[];
  header?: string;
  subHeader?: string;
  moreActions?: ReactNode[];
  hideHeaderDivider?: boolean;
  useDivider?: boolean;
  additionalClasses?: string[];
  additionalHeaderClasses?: string[];
}

const cx = classNames.bind(styles);

export const InfoBox: FunctionComponent<Props> = ({
  items = [],
  header,
  subHeader,
  moreActions = [],
  hideHeaderDivider,
  useDivider,
  additionalClasses = [],
  additionalHeaderClasses = []
}: Props) => {
  const listItems = items.map((item, i) => (
    <div key={i} className={cx({ useDivider, item: true })}>
      {item}
    </div>
  ));
  const headerDivide = !hideHeaderDivider ? items.length : undefined;

  return (
    <div className={[styles.infoBox, ...additionalClasses].join(' ')}>
      {header && (
        <div className={cx({ topContainer: true, headerDivide })}>
          <div className={styles.headerContainer}>
            <div
              className={[
                cx({
                  header: true,
                  withSubHeader: subHeader
                }),
                ...additionalHeaderClasses
              ].join(' ')}
            >
              {header}
            </div>
            {!!moreActions.length && <div className={styles.moreAction}>{...moreActions}</div>}
          </div>
          {subHeader && <div className={styles.subHeader}>{subHeader}</div>}
        </div>
      )}

      <div className={styles.itemList}>{listItems}</div>
    </div>
  );
};

InfoBox.displayName = 'InfoBox';
export default InfoBox;
