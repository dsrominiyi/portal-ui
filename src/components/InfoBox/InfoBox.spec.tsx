import { storyRenders } from './InfoBox.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('InfoBox', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
