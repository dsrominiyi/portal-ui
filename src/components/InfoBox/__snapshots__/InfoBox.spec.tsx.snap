// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox component story renders match snapshots: noDivider 1`] = `
<div
  className="infoBox"
>
  <div
    className="topContainer headerDivide"
  >
    <div
      className="headerContainer"
    >
      <div
        className="header"
      >
        Policies
      </div>
    </div>
  </div>
  <div
    className="itemList"
  >
    <div
      className="item"
      key="0"
    >
      <PolicyWidget
        id="PN0000"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="1"
    >
      <PolicyWidget
        id="PN0001"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="2"
    >
      <PolicyWidget
        id="PN0002"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="3"
    >
      <PolicyWidget
        id="PN0003"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="4"
    >
      <PolicyWidget
        id="PN0004"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="5"
    >
      <PolicyWidget
        id="PN0005"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="6"
    >
      <PolicyWidget
        id="PN0006"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="7"
    >
      <PolicyWidget
        id="PN0007"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="8"
    >
      <PolicyWidget
        id="PN0008"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="9"
    >
      <PolicyWidget
        id="PN0009"
        type="pc"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBox component story renders match snapshots: noHeaderDivider 1`] = `
<div
  className="infoBox"
>
  <div
    className="topContainer"
  >
    <div
      className="headerContainer"
    >
      <div
        className="header"
      >
        Policies
      </div>
    </div>
  </div>
  <div
    className="itemList"
  >
    <div
      className="useDivider item"
      key="0"
    >
      <PolicyWidget
        id="PN0000"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="1"
    >
      <PolicyWidget
        id="PN0001"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="2"
    >
      <PolicyWidget
        id="PN0002"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="3"
    >
      <PolicyWidget
        id="PN0003"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="4"
    >
      <PolicyWidget
        id="PN0004"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="5"
    >
      <PolicyWidget
        id="PN0005"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="6"
    >
      <PolicyWidget
        id="PN0006"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="7"
    >
      <PolicyWidget
        id="PN0007"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="8"
    >
      <PolicyWidget
        id="PN0008"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="9"
    >
      <PolicyWidget
        id="PN0009"
        type="pc"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBox component story renders match snapshots: noItems 1`] = `
<div
  className="infoBox"
>
  <div
    className="topContainer"
  >
    <div
      className="headerContainer"
    >
      <div
        className="header"
      >
        Policies
      </div>
    </div>
  </div>
  <div
    className="itemList"
  />
</div>
`;

exports[`InfoBox component story renders match snapshots: noTitle 1`] = `
<div
  className="infoBox"
>
  <div
    className="itemList"
  >
    <div
      className="item"
      key="0"
    >
      <PolicyWidget
        id="PN0000"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="1"
    >
      <PolicyWidget
        id="PN0001"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="2"
    >
      <PolicyWidget
        id="PN0002"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="3"
    >
      <PolicyWidget
        id="PN0003"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="4"
    >
      <PolicyWidget
        id="PN0004"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="5"
    >
      <PolicyWidget
        id="PN0005"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="6"
    >
      <PolicyWidget
        id="PN0006"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="7"
    >
      <PolicyWidget
        id="PN0007"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="8"
    >
      <PolicyWidget
        id="PN0008"
        type="pc"
      />
    </div>
    <div
      className="item"
      key="9"
    >
      <PolicyWidget
        id="PN0009"
        type="pc"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBox component story renders match snapshots: subHeaderAndMoreActions 1`] = `
<div
  className="infoBox"
>
  <div
    className="topContainer headerDivide"
  >
    <div
      className="headerContainer"
    >
      <div
        className="header withSubHeader"
      >
        Policies
      </div>
      <div
        className="moreAction"
      >
        <Component
          key="moreActionsLink"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="subHeader"
    >
      Private Car
    </div>
  </div>
  <div
    className="itemList"
  >
    <div
      className="useDivider item"
      key="0"
    >
      <PolicyWidget
        id="PN0000"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="1"
    >
      <PolicyWidget
        id="PN0001"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="2"
    >
      <PolicyWidget
        id="PN0002"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="3"
    >
      <PolicyWidget
        id="PN0003"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="4"
    >
      <PolicyWidget
        id="PN0004"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="5"
    >
      <PolicyWidget
        id="PN0005"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="6"
    >
      <PolicyWidget
        id="PN0006"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="7"
    >
      <PolicyWidget
        id="PN0007"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="8"
    >
      <PolicyWidget
        id="PN0008"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="9"
    >
      <PolicyWidget
        id="PN0009"
        type="pc"
      />
    </div>
  </div>
</div>
`;

exports[`InfoBox component story renders match snapshots: titleAndDivider 1`] = `
<div
  className="infoBox"
>
  <div
    className="topContainer headerDivide"
  >
    <div
      className="headerContainer"
    >
      <div
        className="header"
      >
        Policies
      </div>
    </div>
  </div>
  <div
    className="itemList"
  >
    <div
      className="useDivider item"
      key="0"
    >
      <PolicyWidget
        id="PN0000"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="1"
    >
      <PolicyWidget
        id="PN0001"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="2"
    >
      <PolicyWidget
        id="PN0002"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="3"
    >
      <PolicyWidget
        id="PN0003"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="4"
    >
      <PolicyWidget
        id="PN0004"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="5"
    >
      <PolicyWidget
        id="PN0005"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="6"
    >
      <PolicyWidget
        id="PN0006"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="7"
    >
      <PolicyWidget
        id="PN0007"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="8"
    >
      <PolicyWidget
        id="PN0008"
        type="pc"
      />
    </div>
    <div
      className="useDivider item"
      key="9"
    >
      <PolicyWidget
        id="PN0009"
        type="pc"
      />
    </div>
  </div>
</div>
`;
